charts/nextcloud/templates/deployment.yaml

```diff
@@ -305,7 +304,27 @@ spec:
           - "-c"
           - "until pg_isready -h ${POSTGRES_HOST} -U ${POSTGRES_USER} ; do sleep 2 ; done"
       {{- end }}
-      {{- end }}
+      - name: {{ .Chart.Name }}-upgrade
+        image: {{ include "nextcloud.image" . }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+          # Do upgrade and then exit
+          # This avoids doing it in the container, which could get killed by failed liveness probes
+          - "true"
+        env:
+        - name: NEXTCLOUD_UPDATE
+          value: "1"
+{{- include "nextcloud.env" . | indent 8 }}
+        resources:
+{{ toYaml .Values.resources | indent 10 }}
+        {{- if .Values.nextcloud.securityContext}}
+        securityContext:
+          {{- with .Values.nextcloud.securityContext }}
```

Comment:
Could just be a single with instead of if+with

Reply:
The initContainer is copied from the container with only required changes, I am not cleaning up the existing container at this time. It would only make this patch harder to review.